app/views/helpers/TimelineEvents.scala

```diff
@@ -41,22 +42,23 @@ class TimelineEvents @Inject() (
 ) {
   def apply(submission: Submission)(implicit messages: Messages): Seq[TimelineEvent] = {
 
-    val submissionRequestAction = submission.actions.filter(_.requestType == SubmissionRequest)
-    val sortedNotificationsSummaries =
-      submissionRequestAction.headOption.flatMap(_.notifications).getOrElse(Seq.empty[NotificationSummary]).sorted.reverse
+    val allNotificationSummaries = submission.actions.flatMap { action =>
+      val summaries = action.notifications.fold(Seq.empty[NotificationSummary])(identity)
+      if (action.requestType != CancellationRequest) summaries
+      else summaries ++ Seq(NotificationSummary(UUID.randomUUID, action.requestTimestamp, REQUESTED_CANCELLATION))
+    }.sorted
 
-    val IndexToMatchForUploadFilesContent = sortedNotificationsSummaries.indexWhere(summary =>
-      summary.enhancedStatus == ADDITIONAL_DOCUMENTS_REQUIRED || summary.enhancedStatus == UNDERGOING_PHYSICAL_CHECK
-    )
-    val IndexToMatchForViewQueriesContent = sortedNotificationsSummaries.indexWhere(_.enhancedStatus == QUERY_NOTIFICATION_MESSAGE)
-    val IndexToMatchForFixResubmitContent = sortedNotificationsSummaries.indexWhere(_.enhancedStatus == ERRORS)
+    val notificationSummaries =
+      if (!allNotificationSummaries.exists(_.enhancedStatus == CANCELLED)) allNotificationSummaries
+      else allNotificationSummaries.filterNot(_.enhancedStatus == CUSTOMS_POSITION_GRANTED)
```

Comment:
I read this as: if we have received a CANCELLED notification return all the notifications (including the CUSTOMS_POSITION_GRANTED!) else return all notifications apart from CUSTOMS_POSITION_GRANTED.

I think we should always be filtering out CUSTOMS_POSITION_GRANTED as we never want to show this to the user in any circumstance.

Reply:
If we have **NOT** received a CANCELLED notification return all the notifications
else, If we have received a CANCELLED notification, return all notifications apart from CUSTOMS_POSITION_GRANTED

Reply:
Ah, missed the `!` at the start of the if! Yes but we still should not show the CUSTOMS_POSITION_GRANTED in any circumstances

Reply:
I updated the ticket relatively to not show CUSTOMS_POSITION_GRANTED on the timeline.